cli/cmd/inject.go

```diff
@@ -367,10 +367,13 @@ func (options *proxyConfigOptions) overrideConfigs(configs *cfg.All, overrideAnn
 	}
 
 	if options.dockerRegistry != "" {
-		configs.Proxy.ProxyImage.ImageName = registryOverride(configs.Proxy.ProxyImage.ImageName, options.dockerRegistry)
-		configs.Proxy.ProxyInitImage.ImageName = registryOverride(configs.Proxy.ProxyInitImage.ImageName, options.dockerRegistry)
-		overrideAnnotations[k8s.ProxyImageAnnotation] = configs.Proxy.ProxyImage.ImageName
-		overrideAnnotations[k8s.ProxyInitImageAnnotation] = configs.Proxy.ProxyInitImage.ImageName
+		debugImage := configs.GetProxy().GetDebugImage().GetImageName()
+		if debugImage == "" {
+			debugImage = k8s.DebugSidecarImage
+		}
+		overrideAnnotations[k8s.ProxyImageAnnotation] = overwriteRegistry(configs.GetProxy().GetProxyImage().GetImageName(), options.dockerRegistry)
+		overrideAnnotations[k8s.ProxyInitImageAnnotation] = overwriteRegistry(configs.GetProxy().GetProxyInitImage().GetImageName(), options.dockerRegistry)
+		overrideAnnotations[k8s.DebugImageAnnotation] = overwriteRegistry(debugImage, options.dockerRegistry)
```

Comment:
@adleong Here is the "simplified" usage to overwrite whatever registry was used without looking up the registry flag.

I started down the path of even replacing the `registryOverride` method in `root.go` with this implementation, but unit tests were broken around the `install-cni-plugin.go` so I backed off.  What do you think?  Should the `registryOverride` implementation be changed and fix the test accordingly?  I _feel_ like that's really what we would want; not just overriding if the default registry was being used, but the method description was very specific, so possibly very much intended.

Reply:
I'd say it's definitely confusing that there are both `registryOverride` and `overwriteRegistry` methods that do subtly different things.  That said, I don't want this PR to creep in scope forever.  I think my ideal solution would be to go ahead with this and then do a followup PR in which we can take a hard look at `registryOverride` and figure out why it's like that and find a way to unify or differentiate the two functions.

Reply:
Agreed.